cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -97,7 +97,13 @@ protected void checkSystemServices() throws ServiceUnavailableException {
     Callable<Boolean> monitorCallable = new Callable<Boolean>() {
       @Override
       public Boolean call() throws Exception {
-        return getMonitorClient().allSystemServicesOk();
+        // first wait for all system services to be 'OK'
+        if (!getMonitorClient().allSystemServicesOk()) {
+          return false;
+        }
+        // check that the dataset service is up
+        getNamespaceClient().list();
```

Comment:
Its very difficult to verify this fix. Do you think the following would help:

```
try {
  getNamespaceClient().list();
  LOG.info("Able to discover dataset service");
  return true;
} catch (IOException e) {
  LOG.error("Dataset Service may not be up yet");
  return false;
}
```

If there is a better way of making sure that the ``IOException`` is indeed an 'Unable to discover dataset service', that's even better, but I think it may not be super important.